1133 | Header + Footer HCM

chris-lawton opened the pull request:
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1230241133)

### Description of Changes Made
- Ensure mobile menu toggle is visible in high contrast mode in light / dark colour themes
- Ensure main logo is visible in high contrast mode in light / dark colour themes
- Invert footer logos in high contrasr dark mode 
- Ensure footer social icons are visible in high contrast dark mode

### How to Test
- Visit the home page locally
- Enable `forced-colors: active` for HCM and `prefers-color-scheme` for light/dark mode

### Screenshots

<details>
  <summary>Expand to see more</summary>

![Screenshot 2023-07-27 at 14 55 56](https://github.com/torchbox/wagtail-torchbox/assets/31627284/36778a46-8fed-4242-98dc-a6b564fe8024)

![Screenshot 2023-07-27 at 14 56 04](https://github.com/torchbox/wagtail-torchbox/assets/31627284/bd2b1b10-6799-43fa-9a94-245ba5ac9661)

![Screenshot 2023-07-27 at 14 56 12](https://github.com/torchbox/wagtail-torchbox/assets/31627284/e5440e0f-3480-435d-a869-60fea8bc857e)

![Screenshot 2023-07-27 at 14 56 17](https://github.com/torchbox/wagtail-torchbox/assets/31627284/6e151565-cabf-4adc-aa07-7d152d7440ae)

![Screenshot 2023-07-27 at 14 56 33](https://github.com/torchbox/wagtail-torchbox/assets/31627284/47a356be-bb7b-4b85-b559-792158454180)

![Screenshot 2023-07-27 at 14 56 46](https://github.com/torchbox/wagtail-torchbox/assets/31627284/1c78a331-e350-461e-8996-f19ac42078b7)

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Consider adding unit tests, especially for bug fixes. If you don't, tell us why.
- [ ] Tests and linting passes.
- [ ] Consider updating documentation. If you don't, tell us why.
- [ ] If relevant, list the environments / browsers in which you tested your changes.

helenb reviewed (changes requested):
### Issues spotted when testing

**With 'prefers colour scheme: dark'**
- No hover or focus state for the mobile menu button
- No hover state for footer links

**With 'prefers colour scheme: light'**
- No hover or focus state for the mobile menu button
- No hover state for footer links

chris-lawton commented:
Thanks Helen.

I've addressed the hover/focus state for the menu button on https://github.com/torchbox/wagtail-torchbox/pull/513 (oddly the pushed work isn't showing for some reason atm).

I've added an underline and hover/focus state to the footer links.

I'll tackle the proposition page menu on https://github.com/torchbox/wagtail-torchbox/pull/512
